enhancements/validate-catalog-for-bad-bundles.md

```diff
@@ -0,0 +1,177 @@
+
+
+## Motivation
+
+- Provide static validation for [SAP Gardner objects](https://github.com/operator-framework/operator-lifecycle-manager/pull/1564/files?short_path=de0d44a#diff-de0d44a4ac4cb616e43a9a106cdf5701) that would be packaged as part of the [operator bundle](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md). 
+- The tool could be used by RedHat operators and possibly IBM operators
+
+
+### Goals
+
+- Define a set of rules for validating a [operator bundle](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md) before it gets installed on to the cluster. Following resources that are packaged inside a bundle are considered:
+    - [Pod Disruption Budget (PDB)](https://kubernetes.io/docs/tasks/run-application/configure-pdb/#specifying-a-poddisruptionbudget) 
+    - [PriorityClass](https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/#priorityclass)
+    - [VerticalPodAutoScaler](https://cloud.google.com/kubernetes-engine/docs/concepts/verticalpodautoscaler)
+- Design a tool that lists all "bad" bundles for a given [catalog source](https://olm.operatorframework.io/docs/concepts/crds/catalogsource/)
+
+
+### Non-Goals
+
+- Server-side validation of Kube manifests as part of a controller
+
+## Open Questions
+    
+1. Should the tool provide an option to choose a container tool (`--image-builder` flag)?
+2. Potential changes to Scorecard?
+
+## Static Validation Rules
+
+### PDB yaml validation
+Example PDB.yaml
+
+```yaml
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: zk-pdb
+spec:
+  minAvailable: 2  
+  selector:
+    matchLabels:
+      app: zookeeper
+```
+- Either `minAvailable` or `maxUnavailable` should be included, but not both
+- `minAvailable` field cannot be set to 0 or less
+- `maxUnavailable` cannot be set to greater than the number of pods (TODO: check with Dan)
+
+
+### PriorityClass yaml validation
+Example PriorityClass.yaml
+```yaml
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: high-priority
+value: 1000000
+globalDefault: false
+description: "This priority class should be used for XYZ service pods only."
+```
+- `globalDefault` should always be set to false
+
+
+### VerticalPodAutoScaler yaml validation
+Example VerticalPodAutoScaler.yaml
+```yaml
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: my-rec-vpa
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind:       Deployment
+    name:       my-rec-deployment
+  updatePolicy:
+    updateMode: "Off"
+```
+- 
+
+## Proposal
+
+The design proposal has two parts:
+- The first discusses potential approaches for validating SAP Gardener objects.
+- The second part discusses design approach for having a tool to list "bad" bundles in a catalog
+
+## Part 1: Validate bundle objects
```

Comment:
I like the approach of listing different options here, since it can help to give context to the argument., but I would also like to see a definitive statement about which thing this enhancement is ultimately proposing.

Reply:
Updated it to reflect that adding static validation using `golang` is the way forward (with CUE as a potential alternative)